packages/app/components/Questions/Queue/JoinTagGroupButton.tsx

```diff
@@ -0,0 +1,220 @@
+              newQuestionText,
+              [],
+              studentDemo.groupable,
+              studentDemo.isTaskQuestion,
+              studentDemo.location,
+            )
+          }
+        } else {
+          console.log(
+            'Error: This component was used incorrectly. No questionType or taskId provided',
+          )
+        }
+      } else {
+        // If leaving the tag group, remove the questionType (for tag) or remove off the task from the question text
+        if (questionType) {
+          if (!studentQuestion) {
+            console.log('Error: Student does not have a question to leave')
+            return
+          }
+          if (
+            studentQuestion.text === '' &&
+            studentQuestion.questionTypes.length === 1
+          ) {
+            await leaveQueue(false)
+          } else {
+            const newQuestionTypes = studentQuestion.questionTypes.filter(
+              (qt) => qt.id !== questionType.id,
+            )
+            await updateQuestion(
+              studentQuestion.text,
+              newQuestionTypes,
+              studentQuestion.groupable,
+              studentQuestion.isTaskQuestion,
+              studentQuestion.location,
+            )
+          }
+        } else if (taskId) {
+          if (!studentDemo) {
+            console.log('Error: Student does not have a task to leave')
+            return
+          }
+          if (studentDemo.text === `Mark "${taskId}"`) {
+            await leaveQueue(true)
+          } else {
+            const newQuestionText = studentDemo.text.replace(` "${taskId}"`, '')
+            await updateQuestion(
+              newQuestionText,
+              [],
+              studentDemo.groupable,
+              studentDemo.isTaskQuestion,
+              studentDemo.location,
+            )
+          }
+        } else {
+          console.log(
+            'Error: This component was used incorrectly. No questionType or taskId provided',
+          )
+        }
+      }
+      setIsJoined(!isJoined)
+    } catch (e) {
+      message.error(e.response?.data?.message || 'An unexpected error occurred')
+    }
+    setIsLoading(false)
+  }
+
+  return (
+    <CustomButton
+      size="small"
+      className={isDone ? '' : 'join-or-leave-tag-group-button'}
+      colorstyle={isDone ? 'Done' : isJoined ? 'Leave' : 'Join'}
+      onClick={onClick}
+      disabled={beingHelped || isDone || disabled}
+      loading={isLoading}
+      icon={
+        isDone ? (
+          <Check size={15} aria-hidden="true" className="mr-1 shrink-0" />
+        ) : isJoined ? (
+          <LogOut
+            size={13}
+            aria-hidden="true"
+            style={{ transform: 'rotate(180deg)' }}
+            className="mr-1 shrink-0"
+          />
+        ) : (
+          <LogIn size={15} aria-hidden="true" className="mr-1 shrink-0" />
+        )
+      }
+    >
+      {isDone ? 'Done' : isJoined ? 'Leave' : 'Join'}
```

Comment:
Repeats same condition as above

Reply:
This one is for displayed text, the other is for providing a string for specifying why styles to use.